src/core/shape_inference/include/squeeze_shape_inference.hpp

```diff
@@ -71,7 +182,9 @@ std::vector<TRShape> shape_infer(const Squeeze* op,
         NODE_VALIDATION_CHECK(op, false);
     }
 
-    if (arg_rank.is_static() && (unique_axes != nullptr)) {
+    if (!arg_rank.is_static() || (unique_axes == nullptr) || apply_allow_axis_skip(op, unique_axes, arg_shape)) {
```

Comment:
Can be code duplication avoided as shape infer from v0 and v15 is different in this part only?

Reply:
will be done at separate PR with refactor.